sdk/sdk-go/internal/selfBackendVerifier.go

```diff
@@ -0,0 +1,598 @@
+	for _, char := range s {
+		if (char >= 'a' && char <= 'f') || (char >= 'A' && char <= 'F') {
+			return true
+		}
+	}
+	return false
+}
+
+type ConfigIssue struct {
+	Type    ConfigMismatch `json:"type"`
+	Message string         `json:"message"`
+}
+
+type ConfigMismatchError struct {
+	Issues []ConfigIssue `json:"issues"`
+}
+
+func (e *ConfigMismatchError) Error() string {
+	var message []string
+	for _, issue := range e.Issues {
+		message = append(message, fmt.Sprintf("[%s]: %s", issue.Type, issue.Message))
+	}
+	return strings.Join(message, "\n")
+}
+
+func NewConfigMismatchError(issue []ConfigIssue) *ConfigMismatchError {
+	return &ConfigMismatchError{Issues: issue}
+}
+
+type ConfigStore interface {
+	GetConfig(ctx context.Context, id string) (types.VerificationConfig, error)
+	SetConfig(ctx context.Context, id string, config types.VerificationConfig) (bool, error)
+	GetActionId(ctx context.Context, userIdentifier string, actionId string) (string, error)
+}
+
+type SelfBackendVerifier struct {
+	scope                           string
+	identityVerificationHubContract *bindings.IdentityVerificationHubImpl
+	configStorage                   ConfigStore
+	provider                        *ethclient.Client
+	allowedIDs                      map[types.AttestationId]bool
+	userIdentifierType              types.UserIDType
+}
+
+// NewSelfBackendVerifier creates a new SelfBackendVerifier instance
+func NewSelfBackendVerifier(
+	scope string,
+	endpoint string,
+	mockPassport bool,
+	allowedIds map[types.AttestationId]bool,
+	configStorage ConfigStore,
+	userIdentifierType types.UserIDType,
+) (*SelfBackendVerifier, error) {
+	rpcUrl := CELO_MAINNET_RPC_URL
+	hubAddress := IDENTITY_VERIFICATION_HUB_ADDRESS
+
+	if mockPassport {
+		rpcUrl = CELO_TESTNET_RPC_URL
+		hubAddress = IDENTITY_VERIFICATION_HUB_ADDRESS_STAGING
+	}
+
+	provider, err := ethclient.Dial(rpcUrl)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to ethereum client: %v", err)
+	}
+
+	// Create the contract binding
+	hubContract, err := bindings.NewIdentityVerificationHubImpl(
+		common.HexToAddress(hubAddress),
+		provider,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create hub contract binding: %v", err)
+	}
+
+	hashedScope, err := commonUtils.HashEndpointWithScope(endpoint, scope)
+	if err != nil {
+		return nil, fmt.Errorf("failed to hash endpoint with scope: %v", err)
+	}
+
+	return &SelfBackendVerifier{
+		scope:                           hashedScope,
+		identityVerificationHubContract: hubContract,
+		configStorage:                   configStorage,
+		provider:                        provider,
+		allowedIDs:                      allowedIds,
+		userIdentifierType:              userIdentifierType,
+	}, nil
+}
+
+// Verify performs the verification of attestation with the given proof and signals
+func (s *SelfBackendVerifier) Verify(
+	ctx context.Context,
+	attestationId types.AttestationId,
+	proof types.VcAndDiscloseProof,
+	pubSignals []*big.Int,
```

Comment:
accept this in string too. When you're making an API you'll get the inputs in string. Convert them to bigint internally